resources/lib/classes.py

```diff
@@ -0,0 +1,149 @@
+import urlparse
+import urllib
+import unicodedata
+
+
+class genre(object):
+    def __init__(self):
+        self.fanart = None
+        self.thumb = None
+        self.title = None
+        self.genre_slug = None
+
+    def make_kodi_url(self):
+        d = self.__dict__
+        url = ''
+        if d['thumb']:
+            d['thumb'] = urllib.quote_plus(d['thumb'])
+        if d['fanart']:
+            d['fanart'] = urllib.quote_plus(d['fanart'])
+        if d['genre']:
+            d['genre'] = urllib.quote_plus(d['genre'])
+        for item in d.keys():
+            url += '&{0}={1}'.format(item, d[item])
+        return url
+
+    def parse_kodi_url(self, url):
+        params = urlparse.parse_qsl(url)
+        for item in params.keys():
+            setattr(self, item, urllib.unquote_plus(params[item]))
+
+
+class series(object):
+    def __init__(self):
+        self.multi_season = None
+        self.fanart = None
+        self.thumb = None
+        self.series_name = None
+        self.series_slug = None
+        self.season_name = None
+        self.season_slug = None
+        self.title = None
+        self.genre = None
+
+    def get_title(self):
+        if self.multi_season:
+            return '{0} {1}'.format(self.series_name, self.season_name)
+        else:
+            return self.series_name
+
+    def make_kodi_url(self):
+        d = self.__dict__
+        for key, value in d.iteritems():
+            if isinstance(value, unicode):
+                d[key] = unicodedata.normalize(
+                    'NFKD', value).encode('ascii', 'ignore')
+        url = ''
+        if d['thumb']:
+            d['thumb'] = urllib.quote_plus(d['thumb'])
+        if d['fanart']:
+            d['fanart'] = urllib.quote_plus(d['fanart'])
+        if d['genre']:
+            d['genre'] = urllib.quote_plus(d['genre'])
+        for item in d.keys():
+            url += '&{0}={1}'.format(item, d[item])
```

Comment:
I reckon we can do it like this:
```
In [1]: import urllib

In [2]: d = {'thumb': 'http://thumb.com/img', 'fanart': 'http://fanart.com/test', 'genre': 'foobar'}

In [3]: urllib.urlencode(d)
Out[3]: 'fanart=http%3A%2F%2Ffanart.com%2Ftest&genre=foobar&thumb=http%3A%2F%2Fthumb.com%2Fimg'
```

No need to quote here, as urllib does it for us, and this whole thing gets way simpler. What do you think?

Reply:
That's awesome, works like a treat!

Reply:
Great. I'll look forward to checking out your final changes and porting them over to the other addons. 

Doing code reviews is a real luxury for us, and I think it makes our code so much better. Sometimes another pair of eyes can make a huge difference.

Reply:
Definitely, half of my stuff would be certifiable spaghetti without your reviews!